src/renderers/shared/stack/reconciler/ReactInstanceType.js

```diff
@@ -12,22 +12,27 @@
 
 'use strict';
 
+import type {ReactElement} from 'ReactElementType';
+
 export type DebugID = number;
 
 export type ReactInstance = {
-  // ReactCompositeComponent
+  // Shared
   mountComponent: any,
+  unmountComponent: any,
+  receiveComponent: any,
+  getName: () => string,
+  getPublicInstance: any,
+  _currentElement: ReactElement,
```

Comment:
Yes, those methods are shared. `mountComponent`/`receiveComponent`/`updateComponent` is the "base" component API, `DOMComponent` and `CompositeComponent` implement it in different ways.